engine/src/test/java/org/camunda/bpm/engine/test/cmmn/timer/TimerEventListenerTest.java

```diff
@@ -46,4 +51,43 @@ public void testTimerJobData(){
     assertTrue((job instanceof TimerEntity));
   }
 
+  @Deployment(resources = {"org/camunda/bpm/engine/test/cmmn/timer/TimerEventListenerTest.testTimerJobData.cmmn"})
+  public void testTimerEventListenerJobDefinition(){
+    CaseInstance ci = createCaseInstanceByKey("case");
+    assertNotNull(ci);
+    CaseDefinition cd = repositoryService.createCaseDefinitionQuery().caseDefinitionKey("case").singleResult();
+
+    System.out.println("Def count: "+managementService.createJobDefinitionQuery().count());
+
+    List<JobDefinition> allJobs = managementService.createJobDefinitionQuery().list();
+    assertNotNull(allJobs);
+    assertTrue(!allJobs.isEmpty());
+    assertTrue(allJobs.size()==1);
+    assertNotNull(allJobs.get(0));
+    assertTrue(allJobs.get(0).getJobType().equals("timer-event-listener"));
+    assertNotNull(allJobs.get(0).getCaseDefinitionKey());
+    assertNotNull(allJobs.get(0).getCaseDefinitionId());
+
+    //redeploy the same thing. to test isDeploymentNew==false case.
```

Comment:
In order to test `isDeploymentNew == false` you have to do the following:

```
processEngineConfiguration.getDeploymentCache().discardCaseDefinitionCache();
caseService.createCaseInstanceByKey("case");
```

The first line remove the deployed case definition from the cache, and the second line redeploy the case definition again, but in that case it will we called with `isDeploymentNew == false`